rules/mapping.go

```diff
@@ -133,28 +136,38 @@ func (mrs *mappingRuleSnapshot) Schema() (*schema.MappingRuleSnapshot, error) {
 
 // MappingRuleView is a human friendly representation of a mapping rule at a given point in time.
 type MappingRuleView struct {
-	ID           string
-	Name         string
-	CutoverNanos int64
-	Filters      map[string]string
-	Policies     []policy.Policy
+	ID                 string
+	Name               string
+	CutoverNanos       int64
+	Filters            map[string]string
+	Policies           []policy.Policy
+	LastUpdatedBy      string
+	LastUpdatedAtNanos int64
 }
 
-func newMappingRuleView(uuid string, mrs mappingRuleSnapshot) MappingRuleView {
-	mrs = mrs.clone()
-	return MappingRuleView{
-		ID:           uuid,
-		Name:         mrs.name,
-		CutoverNanos: mrs.cutoverNanos,
-		Filters:      mrs.rawFilters,
-		Policies:     mrs.policies,
+func newMappingRuleView(mr *mappingRule, snapshotIdx int) (*MappingRuleView, error) {
+	if snapshotIdx < 0 || snapshotIdx > len(mr.snapshots) {
```

Comment:
mr won't be nil, right?

Reply:
Its never used in a place where it can be, but I can add that check.